lib/foreman_maintain/concerns/directory_marker.rb

```diff
@@ -0,0 +1,35 @@
+module ForemanMaintain
+  module Concerns
+    module DirectoryMarker
+      def with_marked_directory(directory)
+        mark_directory(directory)
+        yield
+        unmark_directory(directory)
+      end
+
+      def find_marked_directory(directory)
+        find_dir_containing_file(directory, mark_name)
+      end
+
+      def mark_name
+        cls = self.class.name.split('::').last.downcase
+        ".#{cls}_directory_mark"
+      end
+
+      private
+
+      def unmark_directory(directory)
+        filename = mark_file(directory)
+        File.delete(filename) if File.exist?(filename)
+      end
+
+      def mark_directory(directory)
+        File.open(mark_file(directory), 'a') {}
+      end
+
+      def mark_file(directory)
```

Comment:
Nitpick: it was a bit confusing for me to see `mark_file` and `mark_directory`, as from the name, they would look like they do the same, but one with dir, the other with file. `mark_file_path` would probably address this problem.